connections/ethereum/connection.go

```diff
@@ -33,19 +34,20 @@ type Connection struct {
 	egsApiKey     string
 	egsSpeed      string
 	conn          *ethclient.Client
-	// signer    ethtypes.Signer
-	opts     *bind.TransactOpts
-	callOpts *bind.CallOpts
-	nonce    uint64
-	optsLock sync.Mutex
-	log      log15.Logger
-	stop     chan int // All routines should exit when this channel is closed
+	itxRpc        *rpc.Client
```

Comment:
Does this field necessarily have to be of type *rpc.Client? Even if someone was to utilize ITX wouldn't they be utilizing the same infura client that they were already connecting to? The only place I see the itxRpc being used is as a flag when sending txs, and for the calling [sendRelayTransaction](https://github.com/ChainSafe/ChainBridge/pull/653/files#diff-afefc070309253e0b4eafe81a4367940467258cc3341fe842c67a8b40aa5cd06R307). Is there a reason that this separate rpc client is being used for that send? 

The forwarder client also is passed a [connection.Client()](https://github.com/ChainSafe/ChainBridge/pull/653/files#diff-b47e840408ea8cf539dd389ec041dbedcf6c0859fc0c6062cc68fa7bef108ebeR145) upon initialization which is of type `*ethclient.Client` and not an `itxRpc`. It seems that a user wanting to use ITX with ChainBridge would need to have two eth clients ready rather than just one which just adds more complexity for someone running a relayer.

Reply:
So I had it this way as it was easier for testing. I was using Ganache for my RPC, but was running a separate ITX mock. In prod I agree that in most cases standard RPC will be the same as the ITX rpc, but there may be edge cases like the one I've described.

Yes, the forwarder client needs access to the chain data so needs to be sure it gets a full rpc - see the problem I had above with testing where the mock doesnt provide all RPC methods.

I agree that having two urls in config is confusing and in most cases unnecessary, but I couldn't think of a good way to get around the testing issues. This local test set up was the one that Palm were using as well.

Reply:
In that case it makes sense, and it will be better to keep it how it is right now.